sql/atxs/atxs.go

```diff
@@ -366,28 +434,43 @@ func Add(db sql.Executor, atx *types.VerifiedActivationTx) error {
 		} else {
 			stmt.BindNull(4)
 		}
-		if atx.VRFNonce != nil {
-			stmt.BindInt64(5, int64(*atx.VRFNonce))
+		if nonce != nil {
+			stmt.BindInt64(5, int64(*nonce))
 		} else {
 			stmt.BindNull(5)
 		}
 		stmt.BindBytes(6, atx.SmesherID.Bytes())
-		stmt.BindBytes(7, buf)
-		stmt.BindInt64(8, atx.Received().UnixNano())
-		stmt.BindInt64(9, int64(atx.BaseTickHeight()))
-		stmt.BindInt64(10, int64(atx.TickCount()))
-		stmt.BindInt64(11, int64(atx.Sequence))
-		stmt.BindBytes(12, atx.Coinbase.Bytes())
-		stmt.BindInt64(13, int64(atx.Validity()))
+		stmt.BindInt64(7, atx.Received().UnixNano())
+		stmt.BindInt64(8, int64(atx.BaseTickHeight()))
+		stmt.BindInt64(9, int64(atx.TickCount()))
+		stmt.BindInt64(10, int64(atx.Sequence))
+		stmt.BindBytes(11, atx.Coinbase.Bytes())
+		stmt.BindInt64(12, int64(atx.Validity()))
+		if atx.PrevATXID != types.EmptyATXID {
+			stmt.BindBytes(13, atx.PrevATXID.Bytes())
+		} else {
+			stmt.BindNull(13)
+		}
 	}
 
 	_, err = db.Exec(`
 		insert into atxs (id, epoch, effective_num_units, commitment_atx, nonce,
-			 pubkey, atx, received, base_tick_height, tick_count, sequence, coinbase, validity)
-		values (?1, ?2, ?3, ?4, ?5, ?6, ?7, ?8, ?9, ?10, ?11, ?12, ?13);`, enc, nil)
+			 pubkey, received, base_tick_height, tick_count, sequence, coinbase,
+			 validity, prev_id)
+		values (?1, ?2, ?3, ?4, ?5, ?6, ?7, ?8, ?9, ?10, ?11, ?12, ?13)`, enc, nil)
 	if err != nil {
 		return fmt.Errorf("insert ATX ID %v: %w", atx.ID(), err)
 	}
+
+	enc = func(stmt *sql.Statement) {
+		stmt.BindBytes(1, atx.ID().Bytes())
+		stmt.BindBytes(2, buf)
+	}
+	_, err = db.Exec("insert into atx_blobs (id, atx) values (?1, ?2)", enc, nil)
```

Comment:
It should probably be enforced that these 2 inserts happen atomically. Maybe via taking an `sql.Tx` as the argument?

Reply:
Noted the problem as tech debt in #5844